### PR TITLE
Jekyll 3.0.x

### PR DESCRIPTION
**Pull request by benbalter**
This is a progress tracking branch for getting things where the GitHub Pages Gem could be bumped to use Jekyll 3.0.x and related dependencies. 

Initial todo to establish a baseline of what's changed:

- [x] Get the Gemfile to bundle
- [x] Confirm plugins work on 3.x
  - [x] jemoji             
  - [x] jekyll-mentions 
  - [x] jekyll-redirect-from 
  - [x] jekyll-sitemap 
  - [x] jekyll-feed
- [x] Enumerate breaking changes

## Known breaking changes

* Dropped support for Ruby < 2.0
* Rouge is now the default highlighter
* Coffeescript and paginate must be explicitly included
* Dropped Maruku support

Please contribute only constructive comments below. If you would like to follow the effort's progress, you may click `subscribe` in the right-hand sidebar.

### CODE REVIEW

**Comment by parkr** [comment]
:pray:

**Comment by parkr** [comment]
@benbalter jekyll-redirect-from has been updated: https://github.com/jekyll/jekyll-redirect-from/releases/tag/v0.9.0

**Comment by parkr** [comment]
@benbalter Also, see https://github.com/jekyll/jekyll/issues/3945 for a list of gems that have been CI'd against Jekyll 3.

**Comment by benbalter** [comment]
Thanks @parkr looks like just mentions and jemoji, which have fixes in the works.

**Comment by parkr** [comment]
@benbalter Did you add jekyll-paginate and jekyll-coffeescript?

**Comment by benbalter** [comment]
> Did you add jekyll-paginate?

Added.

File: lib/github-pages.rb [hunk 1]
```diff
@@ -7,29 +7,30 @@ class GitHubPages
   def self.gems
     {
       # Jekyll
-      "jekyll"                => "2.4.0",
+      "jekyll"                => "3.0.0",
       "jekyll-coffeescript"   => "1.0.1",
       "jekyll-sass-converter" => "1.3.0",
+      "jekyll-paginate"       => "1.1.0",
```
Comment (parkr) [inline lib/github-pages.rb]: Both jekyll-paginate and jekyll-coffeescript can be moved down to "Plugins" as they're no longer runtime dependencies of Jekyll. :)

**Comment by parkr** [comment]
Marvelous!

File: lib/github-pages.rb [hunk 2]
```diff
@@ -7,29 +7,31 @@ class GitHubPages
   def self.gems
     {
       # Jekyll
-      "jekyll"                => "2.4.0",
-      "jekyll-coffeescript"   => "1.0.1",
+      "jekyll"                => "3.0.0",
       "jekyll-sass-converter" => "1.3.0",
```
Comment (Stargator) [inline lib/github-pages.rb]: Since jekyll-coffeescript was moved to plugins, shouldn't jekyll-sass-converter be moved as well?

And are the plugins listed in a particular order?

Comment (parkr) [inline lib/github-pages.rb]: jekyll-sass-converter is still a runtime dependency, so we can keep it up here.

Comment (benbalter) [inline lib/github-pages.rb]: Out of personal curiosity, can you explain / link to the decision to differentiate between the pre processors?

Comment (parkr) [inline lib/github-pages.rb]: @benbalter Do you mean why we removed coffeescript and not sass? Two reasons: Sass is very heavily used, whereas CoffeeScript is muuuch less-oft used, and because coffeescript had a hard dependency on NodeJS/therubyracer (see https://github.com/jekyll/jekyll/issues/3324#issuecomment-72342813). So we opted to keep sass-converter because we felt it was used often and it didn't require many extra dependencies so it was a huge win there, too.

Comment (benbalter) [inline lib/github-pages.rb]: Ahhh that's right. Thanks for the context. 😸

**Comment by parkr** [comment]
@benbalter I believe you were waiting for this: https://github.com/jekyll/jekyll/releases/tag/v3.0.1

File: lib/github-pages.rb [hunk 3]
```diff
@@ -7,32 +7,31 @@ class GitHubPages
...
-      "pygments.rb"               => "0.6.3",
+      "rouge"                     => "1.10.1",
 
       # Plugins
-      "jemoji"                    => "0.5.0",
-      "jekyll-mentions"           => "0.2.1",
-      "jekyll-redirect-from"      => "0.8.0",
+      "jemoji"                    => "0.5.1",
+      "jekyll-mentions"           => "1.0.0",
+      "jekyll-redirect-from"      => "0.9.0",
```
Comment (parkr) [inline lib/github-pages.rb]: 0.9.1 released today. Should I make another PR?

https://github.com/jekyll/jekyll-redirect-from/releases/tag/v0.9.1

https://github.com/jekyll/jekyll-redirect-from/compare/v0.9.0...v0.9.1

Comment (Stargator) [inline lib/github-pages.rb]: If it doesn't push back the release of pages-gem, I'd say sure go ahead.

Comment (parkr) [inline lib/github-pages.rb]: @benbalter Would it be possible to see 0.9.1 cut into this release?

**Comment by vdclouis** [comment]
any news on this?
can I help somehow?

File: lib/github-pages.rb (hunk 3 above)
Comment (WasabiFan) [inline lib/github-pages.rb]: :+1: I have a personal incentive to see `v0.9.1` of jekyll-redirect-from included here!

**Comment by parkr** [comment]
@vdclouis No news that I can report. I believe the process requires review from the GitHub Security team, and availability of the Support team as well. As this is the holiday season, they may have chosen to delay this so the Support team could get some much-needed R&R before the New Year. As this is a breaking change, I anticipate some upset.

File: lib/github-pages.rb (hunk 3 above)
Comment (benbalter) [inline lib/github-pages.rb]: @parkr mind submitting a PR to bump to 0.9.1 on master? If tests pass, feel free to merge (and I believe you can release). Would love to keep this PR as small as possible so we have fewer moving parts, and it doesn't look like the patch bump is releated to Jekyll 3 (assuming it's backwards compatible).

**Comment by alexconst** [comment]
Considering the work being done here, would it be worth submitting a fix to [jekyll's issue with (the lack of) bold text on firefox](https://github.com/jekyll/jekyll/issues/4042)?
(And there is also the issue with the header sizes, like h2 font being bigger than h1)
Which would *eventually* bump jekyll to 2.4.1 and would *eventually* be adopted by github.
I have no clue on the progress timeline for these things.


FWIW, test case for both bugs (visible in firefox):
~~~markdown
---
layout: post
title:  "Test case"
date:   2016-01-10 21:44:44
categories: jekyll update
---

# header 1

test

## header 2

test

### header 3

testing **bold text here**

~~~

File: lib/github-pages.rb (hunk 3 above)
Comment (parkr) [inline lib/github-pages.rb]: No problem, @benbalter: #186.

**Comment by j-mes** [comment]
@alexconst Wouldn't this be more of a CSS issue than Jekyll itself?

**Comment by alexconst** [comment]
@j-mes Well, it's the CSS produced by Jekyll using its `site_template`. But I get your point. 
And since the solution is already documented it's not really worthwhile.

**Comment by benbalter** [comment]
Just a quick update for those following along, this effort is still very much on track, and should be available at a GitHub Pages site near you in the coming month.

In the interim, if you'd like to test your site against the GitHub Pages Jekyll 3 stack, see @parkr's excellent post over in https://talk.jekyllrb.com/t/jekyll-3-x-with-github-pages/1832.

**Comment by larryfox** [comment]
@benbalter Is it safe to assume Github pages will be targeting `3.1.x` now, since that's been released in the interim?

**Comment by parkr** [comment]
@larryfox: no, v3.0.1 is currently targeted. 3.1 will come later.

**Comment by benbalter** [comment]
> Is it safe to assume Github pages will be targeting 3.1.x now, since that's been released in the interim?

There are a lot of moving parts in 3.1 (specifically drops), with few user-facing features other than performance. We're focusing on getting folks happy on 3.0.x, and then can come back and take a look at 3.x in another pass. This will also give 3.1 some time to bake in the wild.

**Comment by larryfox** [comment]
Understandable, thanks for the clarification!

**Comment by jordandanielewski** [comment]
I think there is a problem with Kramdown or Rouge:

**Before:**

```ruby
source 'https://rubygems.org'
gem "github-pages"
```

capture d ecran 2016-02-01 a 02 15 05

---

**After:**

```ruby
source 'https://rubygems.org'
gem "github-pages", github: "github/pages-gem", branch: "jekyll-3"
```

capture d ecran 2016-02-01 a 02 13 57

---

I use [Prismjs](http://prismjs.com/) for the highlighting, so it is based on classes of `<pre>`
The highliting is now broken because the language is not being used in the class

I use

```yaml
kramdown:
  input: GFM
```

capture d ecran 2016-02-01 a 02 20 35

**Comment by vdclouis** [comment]
w00p!

**Comment by pathawks** [comment]
:tada: :+1:
